src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Connections/TaskSocketAsyncEventArgs.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Sources;
+
+namespace System.Net.Connections
+{
+    internal sealed class TaskSocketAsyncEventArgs : SocketAsyncEventArgs, IValueTaskSource
```

Comment:
Is there really any benefit to this anymore? Why can't we just use Task-based async methods on Socket directly?

If there is some benefit, it would be nice to understand what it is, and fix the Task-based methods if possible.

Reply:
This is used to get access at some features not yet exposed via TAP methods -- like cancellable connect and gathered writes.

Reply:
It's super lame that we don't expose cancellable connect via TAP. Is there an issue on this?

Gathered write would be good to do as well, but it's a bit more complicated so it's understandable why we don't have it currently.

Reply:
https://github.com/dotnet/runtime/issues/33418 and https://github.com/dotnet/runtime/issues/33417